frontend/src/reducer/reducer.js

```diff
@@ -1,9 +1,44 @@
 import initialState from "../store/store";
 
-const Reducer = (state = initialState, action) => {
-  const newState = state;
+const reducer = (state = initialState, action) => {
+  const newState = JSON.parse(JSON.stringify(state));
 
-  return newState;
+  switch (action.type) {
+    case "UPDATE_KM_COUNTER":
+      const kmToUpdate = newState.kmToUpdate;
+
+      switch (action.value) {
+        case "erase":
```

Comment:
I think this is bugged: You could enter the case (`action.value==='erase'`) and *not* enter the if (`numOfKmUpdates<6`). In this specific case: you wouldn't do anything but would not leave the switch, leading to the execution of all other cases ! :scream: 
More seriously, just get the `break` out of the `if`, just as you did in the lower case ;-)